<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Measurement and Partitioning of Similarity-Sensitive Biodiversity • rdiversity</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Measurement and Partitioning of Similarity-Sensitive Biodiversity">
<meta property="og:description" content="Provides a framework for the measurement and partitioning of
    the (similarity-sensitive) biodiversity of a metacommunity and its
    constituent subcommunities. Richard Reeve, et al. (2016) 
    &lt;arXiv:1404.6520v3&gt;.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">rdiversity</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/boydorr/rdiversity/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="rdiversity-diversity-measurement-in-r" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#rdiversity-diversity-measurement-in-r" class="anchor"></a>rdiversity: diversity measurement in R</h1></div>

<p>Note that the following documentation applies only to the development version of the package, which can be installed using: <code><a href="https://devtools.r-lib.org//reference/remote-reexports.html">devtools::install_github("boydorr/rdiversity")</a></code>.</p>
<p><code>rdiversity</code> is a package for R based around a framework for measuring biodiversity using similarity-sensitive diversity measures. It provides functionality for measuring alpha, beta and gamma diversity of metacommunities (<em>e.g.</em> ecosystems) and their constituent subcommunities, where similarity may be defined as taxonomic, phenotypic, genetic, phylogenetic, functional, and so on. It uses the diversity framework described in the arXiv paper <a href="https://arxiv.org/abs/1404.6520">arXiv:1404.6520 (q-bio.QM)</a>, “How to partition diversity”.</p>
<p>This package has now reached a stable release and is cross-validated against our Julia package <a href="https://github.com/richardreeve/Diversity.jl">Diversity.jl</a>, which is developed independently. Please <a href="https://github.com/boydorr/rdiversity/issues">raise an issue</a> if you find any problems.</p>
<div id="contents" class="section level2">
<h2 class="hasAnchor">
<a href="#contents" class="anchor"></a>Contents</h2>
<p>Getting started:</p>
<ul>
<li><a href="#installation">Installation</a></li>
<li><a href="#generating-a-metacommunity">Generating a metacommunity</a></li>
<li><a href="#calculating-diversity">Calculating diversity</a></li>
</ul>
<p>Special cases:</p>
<ul>
<li><a href="#taxonomic-diversity">Taxonomic diversity</a></li>
<li><a href="#phylogenetic-diversity">Phylogenetic diversity</a></li>
<li><a href="#genetic-diversity">Genetic diversity</a></li>
<li><a href="#user-defined-distance">User defined distance</a></li>
<li><a href="#user-defined-similarity">User defined similarity</a></li>
</ul>
<p>Miscellaneous:</p>
<ul>
<li><a href="#additional-tools">Additional tools</a></li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install rdiversity from CRAN, simply run the following from an R console:</p>
<pre class="{r}"><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("rdiversity")</a></code></pre>
</div>
<div id="generating-a-metacommunity" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-a-metacommunity" class="anchor"></a>Generating a metacommunity</h2>
<p>Before calculating diversity a <code>metacommunity</code> object must be created. This object contains all the information needed to calculate diversity. In the following example, we generate a metacommunity (<code>partition</code>) comprising two species (“cows” and “sheep”), and partitioned across three subcommunities (a, b, and c).</p>
<pre class="{r}"><code><a href="https://rdrr.io/pkg/rdiversity/man"># Load the package into R
library(rdiversity)

# Initialise data
partition &lt;- data.frame(a=c(1,1),b=c(2,0),c=c(3,1))
row.names(partition) &lt;- c("cows", "sheep")</a></code></pre>
<p>The <code><a href="reference/metacommunity-methods.html">metacommunity()</a></code> function takes two arguments, <code>partition</code> and <code>similarity</code>. When species are considered completely distinct, an identity matrix is required, which is generated automatically if the <code>similarity</code> argument is missing, as below:</p>
<pre class="{r}"><code># Generate metacommunity object
meta &lt;- metacommunity(partition = partition)</code></pre>
<p>Note that a warning is displayed when abundances (rather than relative abundances) are entered into the <code>partition</code> argument. Both are acceptable inputs.</p>
<p>When species share some similarity and a similarity matrix is available, then a similarity object (and the metacommunity object) is generated in the following way:</p>
<pre class="{r}"><code># Initialise similarity matrix
s &lt;- matrix(c(1, 0.5, 0.5, 1), nrow = 2)
row.names(s) &lt;- c("cows", "sheep")
colnames(s) &lt;- c("cows", "sheep")

# Generate similarity object 
s &lt;- similarity(similarity = s, dat_id = "my_taxonomic")

# Generate metacommunity object
meta &lt;- metacommunity(partition = partition, similarity = s)</code></pre>
<p>Alternatively, if a distance matrix is available, then a distance object is generated in the following way:</p>
<pre class="{r}"><code># Initialise distance matrix
d &lt;- matrix(c(0, 0.7, 0.7, 0), nrow = 2)
row.names(d) &lt;- c("cows", "sheep")
colnames(d) &lt;- c("cows", "sheep")

# Generate distance object
d &lt;- distance(distance = d, dat_id = "my_taxonomic")

# Convert the distance object to similarity object (by means of a linear or exponential transform)
s &lt;- dist2sim(dist = d, transform = "linear")

# Generate metacommunity object
meta &lt;- metacommunity(partition = partition, similarity = s)</code></pre>
<p>Each <code>metacommunity</code> object contains the following slots:</p>
<ul>
<li>
<code>@type_abundance</code> : the abundance of types within a metacommunity,<br>
</li>
<li>
<code>@similarity</code> : the pair-wise similarity of types within a metacommunity,<br>
</li>
<li>
<code>@ordinariness</code> : the ordinariness of types within a metacommunity,<br>
</li>
<li>
<code>@subcommunity_weights</code> : the relative weights of subcommunities within a metacommunity, and</li>
<li>
<code>@type_weights</code> : the relative weights of types within a metacommunity.</li>
</ul>
</div>
<div id="calculating-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#calculating-diversity" class="anchor"></a>Calculating diversity</h2>
<div id="method-1" class="section level3">
<h3 class="hasAnchor">
<a href="#method-1" class="anchor"></a>Method 1</h3>
<p>This method uses a wrapper function to simplify the pipeline and is recommended if only a few measures are being calculated.</p>
<p>A complete list of these functions is shown below:</p>
<ul>
<li>
<code><a href="reference/raw_sub_alpha.html">raw_sub_alpha()</a></code> : estimate of naive-community metacommunity diversity<br>
</li>
<li>
<code><a href="reference/norm_sub_alpha.html">norm_sub_alpha()</a></code> : similarity-sensitive diversity of subcommunity <em>j</em> in isolation<br>
</li>
<li>
<code><a href="reference/raw_sub_rho.html">raw_sub_rho()</a></code> : redundancy of subcommunity <em>j</em><br>
</li>
<li>
<code><a href="reference/norm_sub_rho.html">norm_sub_rho()</a></code> : representativeness of subcommunity <em>j</em><br>
</li>
<li>
<code><a href="reference/raw_sub_beta.html">raw_sub_beta()</a></code> : distinctiveness of subcommunity <em>j</em><br>
</li>
<li>
<code><a href="reference/norm_sub_beta.html">norm_sub_beta()</a></code> : estimate of effective number of distinct subcommunities<br>
</li>
<li>
<code><a href="reference/sub_gamma.html">sub_gamma()</a></code> : contribution per individual toward metacommunity diversity<br>
</li>
<li>
<code><a href="reference/raw_meta_alpha.html">raw_meta_alpha()</a></code> : naive-community metacommunity diversity<br>
</li>
<li>
<code><a href="reference/norm_meta_alpha.html">norm_meta_alpha()</a></code> : average similarity-sensitive diversity of subcommunities<br>
</li>
<li>
<code><a href="reference/raw_meta_rho.html">raw_meta_rho()</a></code> : average redundancy of subcommunities<br>
</li>
<li>
<code><a href="reference/norm_meta_rho.html">norm_meta_rho()</a></code> : average representativeness of subcommunities<br>
</li>
<li>
<code><a href="reference/raw_meta_beta.html">raw_meta_beta()</a></code> : average distinctiveness of subcommunities<br>
</li>
<li>
<code><a href="reference/norm_meta_beta.html">norm_meta_beta()</a></code> : effective number of distinct subcommunities<br>
</li>
<li>
<code><a href="reference/meta_gamma.html">meta_gamma()</a></code> : metacommunity similarity-sensitive diversity</li>
</ul>
<p>Each of these functions take two arguments, <code>meta</code> (a <code>metacommunity</code> object) and <code>qs</code> (a vector of q values), and output results as a <code>rdiv</code> object. For example, to calculate normalised subcommunity alpha diversity for <em>q=0</em>, <em>q=1</em>, and <em>q=2</em>:</p>
<pre class="{r}"><code># Initialise data
partition &lt;- data.frame(a=c(1,1),b=c(2,0),c=c(3,1))
row.names(partition) &lt;- c("cows", "sheep")

# Generate a metacommunity object
meta &lt;- metacommunity(partition)

# Calculate diversity
norm_sub_alpha(meta, 0:2)</code></pre>
<p>However, if multiple measures are required and computational efficiency is an issue, then the following method is recommended (the same results are obtained).</p>
</div>
<div id="method-2" class="section level3">
<h3 class="hasAnchor">
<a href="#method-2" class="anchor"></a>Method 2</h3>
<p>This method requires that we first calculate the species-level components, by passing a <code>metacommunity</code> object to the appropriate function; <code><a href="reference/raw_alpha.html">raw_alpha()</a></code>, <code><a href="reference/norm_alpha.html">norm_alpha()</a></code>, <code><a href="reference/raw_beta.html">raw_beta()</a></code>, <code><a href="reference/norm_beta.html">norm_beta()</a></code>, <code><a href="reference/raw_rho.html">raw_rho()</a></code>, <code><a href="reference/norm_rho.html">norm_rho()</a></code>, or <code><a href="reference/raw_gamma.html">raw_gamma()</a></code>. Subcommunity- and metacommunity-level diversities are calculated using the functions <code><a href="reference/subdiv.html">subdiv()</a></code> and <code><a href="reference/metadiv.html">metadiv()</a></code>. Since both subcommunity and metacommunity diversity measures are transformations of the same species-level component, this method is computationally more efficient.</p>
<pre class="{r}"><code># Initialise data
partition &lt;- data.frame(a=c(1,1),b=c(2,0),c=c(3,1))
row.names(partition) &lt;- c("cows", "sheep")

# Generate a metacommunity object
meta &lt;- metacommunity(partition)

# Calculate the species-level component for normalised alpha
component &lt;- norm_alpha(meta)

# Calculate normalised alpha at the subcommunity-level 
subdiv(component, 0:2)

# Likewise, calculate normalised alpha at the metacommunity-level 
metadiv(component, 0:2)</code></pre>
<p>In some instances, it may be useful to calculate <strong>all</strong> subcommunity (or metacommunity) measures. In which case, a <code>metacommunity</code> object may be passed directly to <code><a href="reference/subdiv.html">subdiv()</a></code> or <code><a href="reference/metadiv.html">metadiv()</a></code>:</p>
<pre class="{r}"><code><a href="reference/subdiv.html"># Calculate all subcommunity diversity measures
subdiv(meta, 0:2)

# Calculate all metacommunity diversity measures
metadiv(meta, 0:2)</a></code></pre>
</div>
</div>
<div id="taxonomic-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#taxonomic-diversity" class="anchor"></a>Taxonomic diversity</h2>
<ol>
<li>Initialise data:</li>
</ol>
<pre class="{r}"><code># Taxonomic lookup table
Species &lt;- c("tenuifolium", "asterolepis", "simplex var.grandiflora", "simplex var.ochnacea")
Genus &lt;- c("Protium", "Quararibea", "Swartzia", "Swartzia")
Family &lt;- c("Burseraceae", "Bombacaceae", "Fabaceae", "Fabaceae")
Subclass &lt;- c("Sapindales", "Malvales", "Fabales", "Fabales")
lookup &lt;- cbind.data.frame(Species, Genus, Family, Subclass)

# Partition matrix
partition &lt;- matrix(rep(1, 8), nrow = 4)
colnames(partition) &lt;- LETTERS[1:2]
rownames(partition) &lt;- lookup$Species
</code></pre>
<p>and assign values for each taxonomic level:</p>
<pre class="{r}"><code>values &lt;- c(Species = 0, Genus = 1, Family = 2, Subclass = 3, Other = 4)</code></pre>
<ol>
<li>Generate a distance object from a lookup table using the <code><a href="reference/tax2dist.html">tax2dist()</a></code> function:</li>
</ol>
<pre class="{r}"><code>d &lt;- tax2dist(lookup, values)</code></pre>
<p>By default the <code><a href="reference/tax2dist.html">tax2dist()</a></code> argument <code>precompute_dist</code> is TRUE, such that a pairwise distance matrix is calculated automatically and is stored in <code>d@distance</code>. If the taxonomy is too large, <code>precompute_dist</code> can be set to FALSE, which enables pairwise taxonomic similarity to be calculated on the fly, in step 4.</p>
<ol>
<li>Convert the distance object to similarity object (by means of a linear or exponential transform) using the <code><a href="reference/dist2sim.html">dist2sim()</a></code> function:</li>
</ol>
<pre class="{r}"><code>s &lt;- dist2sim(d, "linear")</code></pre>
<ol>
<li>Generate a metacommunity object using the <code><a href="reference/metacommunity-methods.html">metacommunity()</a></code> function:</li>
</ol>
<pre class="{r}"><code>meta &lt;- metacommunity(partition, s)</code></pre>
<ol>
<li>Calculate diversity:</li>
</ol>
<pre class="{r}"><code><a href="reference/meta_gamma.html">meta_gamma(meta, 0:2)</a></code></pre>
</div>
<div id="phylogenetic-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-diversity" class="anchor"></a>Phylogenetic diversity</h2>
<p>Phylogenetic diversity measures can be broadly split into two categories – those that look at the phylogeny as a whole, such as Faith’s (1992) phylogenetic diversity (Faith’s PD), and those that look at pairwise tip distances, such as mean pairwise distance (MPD; Webb, 2000). The framework of measures presented in this package is able to quantify phylogenetic diversity using both of these methods.</p>
<div id="distance-based-phylogenetic-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#distance-based-phylogenetic-diversity" class="anchor"></a>Distance-based phylogenetic diversity</h3>
<ol>
<li>Initialise data:</li>
</ol>
<pre class="{r}"><code># Example data
tree &lt;- ape::rtree(4)
partition &lt;- matrix(1:12, ncol=3)
partition &lt;- partition/sum(partition)</code></pre>
<ol>
<li>Generate a distance matrix using the <code><a href="reference/phy2dist.html">phy2dist()</a></code> function:</li>
</ol>
<pre class="{r}"><code>d &lt;- phy2dist(tree)</code></pre>
<p>By default the <code><a href="reference/phy2dist.html">phy2dist()</a></code> argument <code>precompute_dist</code> is TRUE, such that a pairwise distance matrix is calculated automatically and is stored in <code>d@distance</code>. If the taxonomy is too large, <code>precompute_dist</code> can be set to FALSE, which enables pairwise taxonomic similarity to be calculated on the fly, in step 4.</p>
<ol>
<li>Convert the distance object to similarity object (by means of a linear or exponential transform) using the <code><a href="reference/dist2sim.html">dist2sim()</a></code> function:</li>
</ol>
<pre class="{r}"><code>s &lt;- dist2sim(d, "linear")</code></pre>
<ol>
<li>Generate a metacommunity object using the <code><a href="reference/metacommunity-methods.html">metacommunity()</a></code> function</li>
</ol>
<pre class="{r}"><code>meta &lt;- metacommunity(partition, s)</code></pre>
<ol>
<li>Calculate diversity</li>
</ol>
<pre class="{r}"><code><a href="reference/meta_gamma.html">meta_gamma(meta, 0:2)</a></code></pre>
</div>
<div id="branch-based-phylogenetic-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#branch-based-phylogenetic-diversity" class="anchor"></a>Branch-based phylogenetic diversity</h3>
<ol>
<li>Initialise data:</li>
</ol>
<pre class="{r}"><code>tree &lt;- ape::rtree(4)
partition &lt;- matrix(1:12, ncol=3)
partition &lt;- partition/sum(partition)
colnames(partition) &lt;- letters[1:3]
row.names(partition) &lt;- paste0("sp",1:4)
tree$tip.label &lt;- row.names(partition)</code></pre>
<ol>
<li>Generate a similarity object using the <code><a href="reference/phy2branch.html">phy2branch()</a></code> function</li>
</ol>
<pre class="{r}"><code>s &lt;- phy2branch(tree, partition)</code></pre>
<ol>
<li>Generate a metacommunity object using the <code><a href="reference/metacommunity-methods.html">metacommunity()</a></code> function</li>
</ol>
<pre class="{r}"><code>meta &lt;- metacommunity(partition, s)</code></pre>
<ol>
<li>Calculate diversity</li>
</ol>
<pre class="{r}"><code><a href="reference/meta_gamma.html">meta_gamma(meta, 0:2)</a></code></pre>
<p><strong>Note that</strong>: a metacommunity that was generated using this approach will contain three additional slots:</p>
<ul>
<li>
<code>@raw_abundance</code> : the relative abundance of terminal species (where types are then considered to be historical species),</li>
<li>
<code>@raw_structure</code> : the length of evolutionary history of each historical species</li>
<li>
<code>@parameters</code> : parameters associated with historical species</li>
</ul>
</div>
</div>
<div id="genetic-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#genetic-diversity" class="anchor"></a>Genetic diversity</h2>
<p>Multiallelic data can currently only be dealt with by the GitHub version of rdiversity.</p>
<ol>
<li>Initialise data:</li>
</ol>
<pre class="{r}"><code>library()
vcf_file &lt;- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
vcf &lt;- read.vcfR(vcf_file, verbose = FALSE)
partition &lt;- cbind.data.frame(A = c(rep(1, 9), rep(0, 9)), B = c(rep(0, 9), rep(1, 9)))
partition &lt;- partition/sum(partition)</code></pre>
<ol>
<li>Generate a distance matrix using the <code><a href="reference/gen2dist.html">gen2dist()</a></code> function, data must be of class <code>vcfR</code>:</li>
</ol>
<pre class="{r}"><code>d &lt;- gen2dist(vcf, biallelic = FALSE)</code></pre>
<ol>
<li>Convert the distance object to a similarity object (by means of a linear or exponential transform) using the <code><a href="reference/dist2sim.html">dist2sim()</a></code> function:</li>
</ol>
<pre class="{r}"><code>s &lt;- dist2sim(d, transform = 'l')</code></pre>
<p>Note: the <code><a href="reference/dist2sim.html">dist2sim()</a></code> function contains an optional argument, <code>max_d</code>, which defines the distance at which pairs of individuals have similarity zero. If not supplied this is set to the maximum distance observed in the distance matrix. If comparing different windows on a genome, for example, it is necessary to ensure <code>max_d</code> is the same for each analysis.</p>
<ol>
<li>Generate a metacommunity object using the <code><a href="reference/metacommunity-methods.html">metacommunity()</a></code> function:</li>
</ol>
<pre class="{r}"><code>rownames(partition) &lt;- rownames(s@similarity)
meta &lt;- metacommunity(partition, s)</code></pre>
<ol>
<li>Calculate diversity, for example beta diversity measures are used to identify trends such as differentiation:</li>
</ol>
<pre class="{r}"><code><a href="reference/norm_meta_beta.html">norm_meta_beta(meta, 0:2)</a></code></pre>
<div id="user-defined-distance" class="section level3">
<h3 class="hasAnchor">
<a href="#user-defined-distance" class="anchor"></a>User defined distance</h3>
<pre class="{r}"><code>partition &lt;- matrix(sample(6), nrow = 3)
rownames(partition) &lt;- paste0("sp", 1:3)
partition &lt;- partition / sum(partition)

d &lt;- matrix(c(0,.75,1,.75,0,.3,1,.3,0), nrow = 3)
rownames(d) &lt;- paste0("sp", 1:3)
colnames(d) &lt;- paste0("sp", 1:3)
d &lt;- distance(d, "my_taxonomy")
s &lt;- dist2sim(d, "linear")

meta &lt;- metacommunity(partition, s)</code></pre>
</div>
<div id="user-defined-similarity" class="section level3">
<h3 class="hasAnchor">
<a href="#user-defined-similarity" class="anchor"></a>User defined similarity</h3>
<pre class="{r}"><code>partition &lt;- matrix(sample(6), nrow = 3)
rownames(partition) &lt;- paste0("sp", 1:3)
partition &lt;- partition / sum(partition)

s &lt;- matrix(c(1,.8,0,.8,1,.1,0,.1,1), nrow = 3)
rownames(s) &lt;- paste0("sp", 1:3)
colnames(s) &lt;- paste0("sp", 1:3)
s &lt;- similarity(s, "my_functional")

meta &lt;- metacommunity(partition, s)</code></pre>
</div>
</div>
<div id="additional-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#additional-tools" class="anchor"></a>Additional tools</h2>
<div id="repartition" class="section level3">
<h3 class="hasAnchor">
<a href="#repartition" class="anchor"></a><code>repartition()</code>
</h3>
<pre class="{r}"><code>tree &lt;- ape::rtree(5)
tree$tip.label &lt;- paste0("sp", 1:5)

partition &lt;- matrix(rep(1,10), nrow = 5)
row.names(partition) &lt;- paste0("sp", 1:5)
partition &lt;- partition / sum(partition)
s &lt;- phy2branch(tree, partition)
meta &lt;- metacommunity(partition, s)

new_partition &lt;- matrix(sample(10), nrow = 5)
row.names(new_partition) &lt;- paste0("sp", 1:5)
new_partition &lt;- new_partition / sum(new_partition)

new_meta &lt;- repartition(meta, new_partition)</code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=rdiversity">https://​cloud.r-project.org/​package=rdiversity</a>
</li>
<li>Browse source code at <br><a href="https://github.com/boydorr/rdiversity/">https://​github.com/​boydorr/​rdiversity/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/boydorr/rdiversity/issues">https://​github.com/​boydorr/​rdiversity/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing rdiversity</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Sonia Mitchell <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0003-1536-2066" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Richard Reeve <br><small class="roles"> Author, thesis advisor </small> <a href="https://orcid.org/0000-0003-2589-8091" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/boydorr/rdiversity/actions"><img src="https://github.com/boydorr/rdiversity/workflows/build/badge.svg" alt="test-build"></a></li>
<li><a href="https://codecov.io/gh/boydorr/rdiversity"><img src="https://codecov.io/gh/boydorr/rdiversity/branch/master/graph/badge.svg" alt="codecov"></a></li>
<li><a href="https://doi.org/10.5281/zenodo.597470"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.597470.svg" alt="DOI"></a></li>
<li><a href="http://cranlogs.r-pkg.org/badges/grand-total/rdiversity"><img src="http://cranlogs.r-pkg.org/badges/grand-total/rdiversity?color=yellow" alt="Total downloads"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Sonia Mitchell, Richard Reeve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
